src/topics/TopicEditModal.js

```diff
@@ -0,0 +1,182 @@
+              return _('Topic is required in this stream.');
+            case 'user-did-not-edit':
+              return _("You haven't made any changes.");
+            case 'topic-too-long':
+              return _('Topic too long. Max 60 characters.');
+            default:
+              ensureUnreachable(error);
+              throw new Error();
+          }
+        })
+        .join('\n\n');
+      showErrorAlert(errorMessages);
+      return;
+    }
+    if (streamId !== null && oldTopic !== null && newTopicInputValue !== null) {
+      try {
+        const messageId = await fetchSomeMessageIdForConversation(
+          auth,
+          streamId,
+          oldTopic,
+          streamsById,
+          zulipFeatureLevel,
+        );
+        if (messageId == null) {
+          throw new Error(
+            _('No messages in topic: {streamAndTopic}', {
+              streamAndTopic: `#${streamsById.get(streamId)?.name ?? 'unknown'} > ${oldTopic}`,
+            }),
+          );
+        }
+        await api.updateMessage(auth, messageId, {
+          propagate_mode: 'change_all',
+          subject: newTopicInputValue.trim(),
+          ...(zulipFeatureLevel >= 9 && {
+            send_notification_to_old_thread: true,
+            send_notification_to_new_thread: true,
+          }),
+        });
+      } catch (error) {
+        showErrorAlert('Failed to edit topic.');
+      } finally {
+        closeEditTopicModal();
+      }
+    } else {
+      throw new Error(_('Topic, streamId, or input value null.'));
+    }
+  };
+
+  const modalStyles = createStyleSheet({
+    backdrop: {
+      position: 'absolute',
+      backgroundColor: 'black',
+      opacity: 0.25,
+      top: 0,
+      left: 0,
+      right: 0,
+      bottom: 0,
+    },
+    wrapper: {
+      flex: 1,
+      justifyContent: 'center',
+      alignItems: 'center',
+    },
+    modal: {
+      justifyContent: 'flex-start',
+      backgroundColor,
+      padding: 15,
+      shadowOpacity: 0.5,
+      elevation: 8,
+      shadowRadius: 16,
+      borderRadius: 5,
+      width: 280,
+    },
+    buttonContainer: {
+      flexDirection: 'row',
+      justifyContent: 'flex-end',
+    },
+    titleText: {
+      fontSize: 18,
+      lineHeight: 21,
+      marginBottom: 10,
+      fontWeight: 'bold',
+    },
+  });
+
+  return (
+    <Modal
+      transparent
+      visible={streamId !== null && oldTopic !== null}
+      animationType="fade"
+      onRequestClose={closeEditTopicModal}
+      supportedOrientations={['portrait', 'landscape', 'landscape-left', 'landscape-right']}
+    >
+      <View style={modalStyles.backdrop} />
+      <View style={modalStyles.wrapper}>
+        <View style={modalStyles.modal}>
+          <ZulipTextIntl style={modalStyles.titleText} text="Edit topic" />
+          <Input
+            style={styles.marginBottom}
+            value={newTopicInputValue}
+            placeholder="You may need to enter a topic in this stream."
```

Comment:
Seeing this in the UI, it feels kind of wordy. It also can overflow the input: 
![image](https://user-images.githubusercontent.com/28173/197320699-eff4f19b-1d54-426b-8210-d91c4e072557.png)

Let's go for just "Topic". It is a placeholder, not an instruction, so it doesn't need to say a lot.